<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1109</width>
    <height>756</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>华文中宋</family>
   </font>
  </property>
  <property name="windowTitle">
   <string>生产者消费者Demo</string>
  </property>
  <property name="windowIcon">
   <iconset resource="res.qrc">
    <normaloff>:/icons/assets/icons/logo.png</normaloff>:/icons/assets/icons/logo.png</iconset>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonTextUnderIcon</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QWidget" name="panel_area" native="true">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="sizeConstraint">
            <enum>QLayout::SetDefaultConstraint</enum>
           </property>
           <item>
            <widget class="ProgressBall" name="buffer1" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="cursor">
              <cursorShape>WaitCursor</cursorShape>
             </property>
             <property name="toolTip">
              <string>Buffer1</string>
             </property>
             <property name="whatsThis">
              <string>Buffer1</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lab_buf1">
             <property name="font">
              <font>
               <family>华文中宋</family>
               <pointsize>11</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Buffer1数据量</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <property name="sizeConstraint">
            <enum>QLayout::SetDefaultConstraint</enum>
           </property>
           <item>
            <widget class="ProgressBall" name="buffer2" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="cursor">
              <cursorShape>WaitCursor</cursorShape>
             </property>
             <property name="toolTip">
              <string>Buffer2</string>
             </property>
             <property name="whatsThis">
              <string>Buffer2</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lab_buf2">
             <property name="font">
              <font>
               <family>华文中宋</family>
               <pointsize>11</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Buffer2数据量</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <property name="sizeConstraint">
            <enum>QLayout::SetNoConstraint</enum>
           </property>
           <item>
            <widget class="ProgressBall" name="buffer3" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="cursor">
              <cursorShape>WaitCursor</cursorShape>
             </property>
             <property name="toolTip">
              <string>Buffer3</string>
             </property>
             <property name="whatsThis">
              <string>Buffer3</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lab_buf3">
             <property name="font">
              <font>
               <family>华文中宋</family>
               <pointsize>11</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Buffer3数据量</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QGroupBox" name="groupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>操作日志</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QScrollArea" name="scrollArea1">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents1">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>245</width>
              <height>539</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <item>
              <widget class="QPlainTextEdit" name="plainText_log1">
               <property name="font">
                <font>
                 <family>方正姚体</family>
                 <pointsize>8</pointsize>
                </font>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnClear1">
               <property name="text">
                <string>清屏</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QScrollArea" name="scrollArea2">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>246</width>
              <height>539</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
              <widget class="QPlainTextEdit" name="plainText_log2">
               <property name="font">
                <font>
                 <family>方正姚体</family>
                 <pointsize>8</pointsize>
                </font>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnClear2">
               <property name="text">
                <string>清屏</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QScrollArea" name="scrollArea3">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents3">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>245</width>
              <height>539</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <item>
              <widget class="QPlainTextEdit" name="plainText_log3">
               <property name="font">
                <font>
                 <family>方正姚体</family>
                 <pointsize>8</pointsize>
                </font>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnClear3">
               <property name="text">
                <string>清屏</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1109</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actSetConfig"/>
   <addaction name="separator"/>
   <addaction name="actImportConfig"/>
   <addaction name="actExportResult"/>
   <addaction name="separator"/>
   <addaction name="actStart"/>
   <addaction name="actPasue"/>
   <addaction name="actTerminate"/>
   <addaction name="actRestart"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="windowTitle">
    <string>统计信息</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>系统设置</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_1">
          <property name="text">
           <string>设置ID</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_c_id">
          <property name="text">
           <string>c_id</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Buffer1容量</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label_buffer1_size">
          <property name="text">
           <string>buffer1_size</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Buffer2容量</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="label_buffer2_size">
          <property name="text">
           <string>buffer2_size</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Buffer3容量</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="label_buffer3_size">
          <property name="text">
           <string>buffer3_size</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>put操作数</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLabel" name="label_put_num">
          <property name="text">
           <string>put_num</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_16">
          <property name="text">
           <string>move操作数</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QLabel" name="label_move_num">
          <property name="text">
           <string>move_num</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>get操作数</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QLabel" name="label_get_num">
          <property name="text">
           <string>get_num</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="label_20">
          <property name="text">
           <string>put操作速度</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QLabel" name="label_put_speed">
          <property name="text">
           <string>put_speed</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>move操作速度</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QLabel" name="label_move_speed">
          <property name="text">
           <string>move_speed</string>
          </property>
         </widget>
        </item>
        <item row="9" column="0">
         <widget class="QLabel" name="label_17">
          <property name="text">
           <string>get操作速度</string>
          </property>
         </widget>
        </item>
        <item row="9" column="1">
         <widget class="QLabel" name="label_get_speed">
          <property name="text">
           <string>get_speed</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>系统消息</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QScrollArea" name="scrollArea">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>274</width>
             <height>301</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QPlainTextEdit" name="sysText_log">
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actSetConfig">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/assets/icons/config.png</normaloff>:/icons/assets/icons/config.png</iconset>
   </property>
   <property name="text">
    <string>参数设置</string>
   </property>
   <property name="toolTip">
    <string>参数设置</string>
   </property>
  </action>
  <action name="actImportConfig">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/assets/icons/import.png</normaloff>:/icons/assets/icons/import.png</iconset>
   </property>
   <property name="text">
    <string>导入设置</string>
   </property>
   <property name="toolTip">
    <string>导入设置</string>
   </property>
  </action>
  <action name="actExportResult">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/assets/icons/export.png</normaloff>:/icons/assets/icons/export.png</iconset>
   </property>
   <property name="text">
    <string>导出结果</string>
   </property>
   <property name="toolTip">
    <string>导出结果</string>
   </property>
  </action>
  <action name="actStart">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/assets/icons/start.png</normaloff>:/icons/assets/icons/start.png</iconset>
   </property>
   <property name="text">
    <string>开始</string>
   </property>
   <property name="toolTip">
    <string>开始</string>
   </property>
  </action>
  <action name="actPasue">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/assets/icons/pause.png</normaloff>:/icons/assets/icons/pause.png</iconset>
   </property>
   <property name="text">
    <string>暂停</string>
   </property>
   <property name="toolTip">
    <string>暂停</string>
   </property>
  </action>
  <action name="actTerminate">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/assets/icons/terminate.png</normaloff>:/icons/assets/icons/terminate.png</iconset>
   </property>
   <property name="text">
    <string>结束</string>
   </property>
   <property name="toolTip">
    <string>结束</string>
   </property>
  </action>
  <action name="actRestart">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/assets/icons/restart.png</normaloff>:/icons/assets/icons/restart.png</iconset>
   </property>
   <property name="text">
    <string>重新开始</string>
   </property>
   <property name="toolTip">
    <string>重新开始</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ProgressBall</class>
   <extends>QWidget</extends>
   <header>progressball.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
